plugin-server/src/cdp/hog-executor.ts

```diff
@@ -327,15 +327,51 @@ export class HogExecutor {
         }
     }
 
-    execute(invocation: HogFunctionInvocation): HogFunctionInvocationResult {
+    executeTransformation(invocation: HogFunctionInvocation): HogFunctionInvocationResult {
+        const result = this.execute(invocation, {
+            allowAsync: false,
+            requireResult: true,
+            functionType: 'transformation',
+        })
+
+        // For transformations, we require a result to be returned
+        if (result.finished && result.execResult === undefined) {
+            result.error = 'Transformation function must return a value'
```

Comment:
This isn't true...? A transformation can drop the data by returning null. I think we should leave that side of the checking logic to the parent caller.